Library/Homebrew/test/pkg_version_spec.rb

```diff
@@ -0,0 +1,65 @@
+require "pkg_version"
+
+describe PkgVersion do
+  specify "#parse" do
+    expect(PkgVersion.parse("1.0_1")).to eq(PkgVersion.new(Version.create("1.0"), 1))
+    expect(PkgVersion.parse("1.0_1")).to eq(PkgVersion.new(Version.create("1.0"), 1))
+    expect(PkgVersion.parse("1.0")).to eq(PkgVersion.new(Version.create("1.0"), 0))
+    expect(PkgVersion.parse("1.0_0")).to eq(PkgVersion.new(Version.create("1.0"), 0))
+    expect(PkgVersion.parse("2.1.4_0")).to eq(PkgVersion.new(Version.create("2.1.4"), 0))
+    expect(PkgVersion.parse("1.0.1e_1")).to eq(PkgVersion.new(Version.create("1.0.1e"), 1))
+  end
+
+  specify "#==" do
+    expect(PkgVersion.parse("1.0_0")).to be == PkgVersion.parse("1.0")
+    expect(PkgVersion.parse("1.0_1")).to be == PkgVersion.parse("1.0_1")
+  end
+
+  describe "#>" do
+    it "returns true if the left version is bigger than the right" do
+      expect(PkgVersion.parse("HEAD")).to be > PkgVersion.parse("1.0")
+      expect(PkgVersion.parse("1.1")).to be > PkgVersion.parse("1.0_1")
+    end
+
+    it "raises an error if the other side isn't of the same class" do
+      expect {
+        PkgVersion.new(Version.create("1.0"), 0) > Object.new
+      }.to raise_error(ArgumentError)
+
+      expect {
```

Comment:
I think this should be in its own named example to highlight that `PkgVersion` isn't compatible with `Version`.